### PR TITLE
Update CI configs to v0.4.7

### PR DESCRIPTION
**Pull request by pionbot**
Update lint scripts and CI configs.

### CODE REVIEW

**Comment by scorpionknifes** [comment]
For this one, I've noticed there is a errors.go file. I'm wondering if we need to define the new wrapped errors in here and also if we need to export them.

**Comment by Sean-Der** [comment]
@scorpionknifes I am fine with having the new errors not exported.

I want to make as few changes as possible during the migration. I have been cutting tickets for 'code smells' as I go that we can come back too.

File: rand.go [hunk 1]
```diff
@@ -14,8 +14,10 @@ const (
 // Seeding random generator each time limits number of generated sequence to 31-bits,
 // and causes collision on low time accuracy environments.
 // Use global random generator seeded by crypto grade random.
-var globalMathRandomGenerator = randutil.NewMathRandomGenerator()
-var globalCandidateIDGenerator = candidateIDGenerator{globalMathRandomGenerator}
+var (
+	globalMathRandomGenerator  = randutil.NewMathRandomGenerator()               //nolint:gochecknoglobals
+	globalCandidateIDGenerator = candidateIDGenerator{globalMathRandomGenerator} //nolint:gochecknoglobals
+)
```
Comment (scorpionknifes) [inline rand.go:20]: No idea how to deal with this. Assuming changing this would affect implementation.

**Review by scorpionknifes — COMMENTED** [review]
Complete 😄

**Comment by Sean-Der** [comment]
Amazing work again @scorpionknifes! You are making my life so much easier, thank you so much this.

**Review by Sean-Der — APPROVED** [review]

**Pull request merged**